packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -65,23 +68,50 @@ function HeaderToolbar( { onToggleInserter, isInserterOpen } ) {
 			className="edit-post-header-toolbar"
 			aria-label={ toolbarAriaLabel }
 		>
-			<Button
-				className="edit-post-header-toolbar__inserter-toggle"
-				isPrimary
-				isPressed={ isInserterOpen }
-				onClick={ onToggleInserter }
-				disabled={ ! isInserterEnabled }
-				icon={ plus }
-				label={ _x(
-					'Add block',
-					'Generic label for block inserter button'
+			<ToolbarItem>
+				{ ( itemProps ) => (
+					<Button
+						{ ...itemProps }
+						className="edit-post-header-toolbar__inserter-toggle"
+						isPrimary
+						isPressed={ isInserterOpen }
+						onClick={ onToggleInserter }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={ _x(
+							'Add block',
+							'Generic label for block inserter button'
+						) }
+					/>
+				) }
+			</ToolbarItem>
+			{ isLargeViewport && (
```

Comment:
I'm flagging this for myself to test what happens with focus when you change viewport of your browser and this button disappears when focused. I know it's an edge case and it might be an issue before but I'm simply curious :)

Reply:
I tested it. When one of the blocks were selected before, then focus moves to the selected block 🙃

Reply:
I was wrong, the blinking cursor tricked me, so the focus gets removed, and when you tab it gets restored to one of the items in the toolbar, not that bad I guess. Could Reakit move focus automatically to the next/previous item in such case or would be too far in what it should do? :)

Reply:
In the first iterations, Reakit's `Composite` (and therefore `Toolbar`) automatically moved focus to the next/previous item when the current focused item got unmounted. Later it was changed so the previously **focused** item would get focus, like when you open and close a new tab on Chrome. But I haven't found enough material to support any of those approaches, and this was making the code even more complex for little gain, so I reverted it.

For now, `toolbar.currentId` is still set to the previously focused item, but it doesn't automatically move focus. It's up to the user to decide what to do, but I do plan to revisit it in the future.

Reply:
I would expect it’s going to be tricky to handle 😅 A similar challenge is when you click a button and it should become disabled, like when triggering undo until there is nothing to undo. 

In this particular case, I assume it’s fine to clean up internally Reakit’s state - that is the current value and leave the rest (focus) of handling to the implementer.